packages/extension/src/serviceWorker/index.ts

```diff
@@ -633,6 +999,38 @@ chrome.runtime.onMessage.addListener(async (request) => {
 
   const incomingMessageTabId = request.payload.tabId;
 
+  if ('GET_REST_DATA_FROM_URL' === incomingMessageType) {
+    let allCookies: CookieData[] = [];
+    await Promise.all(
+      Object.entries(frameIdToResourceMap[incomingMessageTabId]).map(
+        async ([key, value]) => {
+          try {
+            //@ts-ignore
+            const { cookies = [] } = await chrome.debugger.sendCommand(
+              { targetId: key },
+              'Network.getCookies',
+              { urls: Array.from(value) }
+            );
+
+            const parsedCookies = parseNetworkCookies(
+              cookies,
+              syncCookieStore?.getTabUrl(incomingMessageTabId) ?? '',
+              cookieDB ?? {},
+              key
+            );
+            if (parsedCookies.length > 0) {
+              allCookies = [...allCookies, ...parsedCookies];
+            }
+          } catch (error) {
+            //Fail silently. There will be only one reason stating target id not found.
+          }
+        }
+      )
+    );
+
+    syncCookieStore?.update(Number(incomingMessageTabId), allCookies);
+  }
+
```

Comment:
The file is becoming too big, let's refactor and break it down.

Reply:
Completed in [b2aa6a4](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/539/commits/b2aa6a4ca7540b5219ae65b345e6ac9548549cb5)